src/services/smart-contracts/actions/core.js

```diff
@@ -106,7 +111,134 @@ function getReadyCampaign(campaign, identity, Dai) {
 	return newCampaign
 }
 
-export async function openChannel({ campaign, account }) {
+async function getChannelsWithBalance({ identityAddr, requester, STATES }) {
+	const channels = await requester
+		.fetch({
+			route: '/campaigns',
+			method: 'GET',
+			queryParams: {
+				status: STATES.join(','),
+			},
+		})
+		.then(res => res.json())
+
+	return Promise.all(
+		channels
+			.map(channel => {
+				const { lastApprovedBalances } = channel.status
+				if (lastApprovedBalances) {
+					const allLeafs = Object.keys(lastApprovedBalances).map(k =>
+						Channel.getBalanceLeaf(k, lastApprovedBalances[k])
+					)
+					const mTree = new MerkleTree(allLeafs)
```

Comment:
I think you will need the `lastApprovedSigs` here; check the latest code in `routineAuthsLoop`